Foundation/include/Poco/Dynamic/VarHolder.h

```diff
@@ -787,6 +823,287 @@ class VarHolderImpl: public VarHolder
 		return typeid(T);
 	}
 
+	void convert(Int8& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToSigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(Int16& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToSigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(Int32& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToSigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(Int64& val) const
```

Comment:
I think that having partial specialisation for signed (and another for unsigned) integer types would simplify the code. The functions are identical except for the type.

Reply:
the problem is these are virtual functions so they become overloads, not overrides

Reply:
True, missed that.